### PR TITLE
Move common test utilities to stand-alone package

### PR DESCRIPTION
**Pull request by tanabarr**
In order to make common utility test routines usable across number of chroma
top-level directories and/or from chroma packages which may not have access to
the full chroma source directory structure, a new chroma package called
chroma-common-test should be created which contains library of common test
utilities.

This package is created from within the chroma-common.spec file and consists of
files within the chroma-common/chroma_common/test/ directory.

  - Move CommandCaptureTestCase to chroma_common/test
  - Create ImlUnitTestCase in chroma_common/test
  - Make AgentUnitTestCase inherit from ImlUnitTestCase
  - Use multiple inheritance for modules that require CommandCaptureTestCase
    and AgentUnitTestCase
  - Create sub-package chroma-common-test in chroma-common.spec
  - Make chroma-manager-integration-tests require chroma-common-test
  - Consolidate test_manage_target.py into test_manage_targets.py
  - Amend and update unit tests
  - Fix copyright notices

### CODE REVIEW

File: chroma-common/chroma-common.spec [hunk 1]
```diff
@@ -1,7 +1,7 @@
 %{!?name: %define name chroma-common}
-%{?!version: %define version %(%{__python2} -c "from chroma_common import version; sys.stdout.write(version())")}
+%{?!version: %define version %(%{__python} -c "from chroma_common import version; sys.stdout.write(version())")}
```
Comment (brianjmurrell) [inline chroma-common/chroma-common.spec]: Was there a particular reason for switching `__python2` to `__python`?

Since we only work with Python 2.x we should be targeting the 2.x version of python.  `__python` will be changed to point at Python 3.x at some point in the future and we need to make sure we are compatible with Python 3.x before falling back to the generic `__python` macros.

File: chroma-common/chroma-common.spec [hunk 2]
```diff
@@ -1,7 +1,7 @@
...
 %{?!package_release: %define package_release 1}
-%{?!python2_sitelib: %define python2_sitelib %(%{__python2} -c "from distutils.sysconfig import get_python_lib; import sys; sys.stdout.write(get_python_lib())")}
+%{?!python_sitelib: %define python_sitelib %(%{__python} -c "from distutils.sysconfig import get_python_lib; import sys; sys.stdout.write(get_python_lib())")}
```
Comment (brianjmurrell) [inline chroma-common/chroma-common.spec]: Ditto.

File: chroma-common/chroma-common.spec [hunk 3]
```diff
@@ -17,19 +17,36 @@ Vendor: Intel(R) Corporation
...
+Requires: %{name} = %{version}-%{release}
+%description test
+This package contains shared test utilities used in the test framework for Manager for Lustre.
+
 %prep
 %setup -n %{name}-%{version}
 
 %build
-%{__python2} setup.py build
+%{__python} setup.py build
```
Comment (brianjmurrell) [inline chroma-common/chroma-common.spec]: Ditto.

File: chroma-common/chroma-common.spec [hunk 4]
```diff
@@ -17,19 +17,36 @@ Vendor: Intel(R) Corporation
...
 
 %install
 rm -rf %{buildroot}
-%{__python2} setup.py install --skip-build --install-lib=%{python2_sitelib} --root=%{buildroot}
+%{__python} setup.py install --skip-build --install-lib=%{python_sitelib} --root=%{buildroot}
```
Comment (brianjmurrell) [inline chroma-common/chroma-common.spec]: Ditto.

File: chroma-common/chroma-common.spec [hunk 5]
```diff
@@ -17,19 +17,36 @@ Vendor: Intel(R) Corporation
...
+  install_file=${test_file/$RPM_BUILD_ROOT\///}
+  echo "${install_file}" >> test.files
+done
+
 %clean
 rm -rf %{buildroot}
 
 %files
-%{python2_sitelib}/*
+%exclude %{python_sitelib}/chroma_common/test
```
Comment (brianjmurrell) [inline chroma-common/chroma-common.spec]: Ditto.

File: chroma-common/chroma-common.spec [hunk 6]
```diff
@@ -17,19 +17,36 @@ Vendor: Intel(R) Corporation
...
-%{__python2} setup.py install --skip-build --install-lib=%{python2_sitelib} --root=%{buildroot}
+%{__python} setup.py install --skip-build --install-lib=%{python_sitelib} --root=%{buildroot}
 mkdir -p $RPM_BUILD_ROOT/usr/sbin/
 
+touch test.files
+for test_file in $(find -L $RPM_BUILD_ROOT%{python_sitelib}/chroma_common/test/ -type f -name '*.py*'); do
+  install_file=${test_file/$RPM_BUILD_ROOT\///}
+  echo "${install_file}" >> test.files
+done
+
```
Comment (brianjmurrell) [inline chroma-common/chroma-common.spec]: Is there any reason we need to do this rather than just adding:

```
%{python2_sitelib}/chroma_common/test
```

to the `%files test` section?

File: chroma-common/chroma-common.spec (hunk 1 above)
Comment (tanabarr) [inline chroma-common/chroma-common.spec]: I made the change to using __python2 in the previous patch but it's causing problems in this patch so I have reverted it to what is currently on master (no change). if we want to move to using __python2 in all the spec files we should do it in a separate patch.

File: chroma-common/chroma-common.spec (hunk 2 above)
Comment (tanabarr) [inline chroma-common/chroma-common.spec]: as above

File: chroma-common/chroma-common.spec (hunk 3 above)
Comment (tanabarr) [inline chroma-common/chroma-common.spec]: as above

File: chroma-common/chroma-common.spec (hunk 4 above)
Comment (tanabarr) [inline chroma-common/chroma-common.spec]: as above

File: chroma-common/chroma-common.spec (hunk 6 above)
Comment (tanabarr) [inline chroma-common/chroma-common.spec]: good point, corrected in corrected in https://github.com/intel-hpdd/intel-manager-for-lustre/pull/76

File: chroma-common/chroma-common.spec (hunk 5 above)
Comment (tanabarr) [inline chroma-common/chroma-common.spec]: as above

File: chroma-common/chroma-common.spec (hunk 1 above)
Comment (brianjmurrell) [inline chroma-common/chroma-common.spec]: What kind of problems was it causing?
